docs/Node-for-Formula-Authors.md

```diff
@@ -102,7 +102,19 @@ class Foo < Formula
 
   def install
     system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
+
+    # "link" the executable `foo`, ensuring that the version of node installed by Homebrew is used
+    # (the created `foo` will set the ENV before exec'ing your executable)
+    env = { PATH: "#{HOMEBREW_PREFIX/"bin"}:$PATH" }
+    (bin/"foo").write_env_script "#{libexec}/bin/foo", env
+
+    # Uncomment if you simply want to symlink the executable – note that this means the first
+    # `node` on the PATH will be used (not necessarily the one Homebrew installed)
+    # bin.install_symlink Dir["#{libexec}/bin/*"]
```

Comment:
Which/how many formulae do this each of these ways? CC @homebrew/core for thoughts.

Reply:
I personally don't know when you'd ever want it this way – it's just how it was documented previously.

[yarn](https://github.com/Homebrew/homebrew-core/blob/1ac27774129258f13ae68865ee511709bbb3e935/Formula/y/yarn.rb#L23) will use the first `node` on the path but I'd imagine this is an outlier (and it is also not calling `bin.install_symlink` – but it is _not_ trying to always use homebrew's node)

Reply:
Looking at the 173 formulae with `require "language/node"`:

5 use `write_env_script`, although only 3 use it in the exact way suggested here:
- [`protoc-gen-grpc-web`](https://github.com/Homebrew/homebrew-core/blob/f4861d055bf1593f5d0b1934b4756cb9ed9e8246/Formula/p/protoc-gen-grpc-web.rb#L33-L34)
  - This doesn't actually use `system "npm", "install", *Language::Node.std_npm_install_args(libexec)` so maybe it shouldn't count
- [`mongosh`](https://github.com/Homebrew/homebrew-core/blob/f4861d055bf1593f5d0b1934b4756cb9ed9e8246/Formula/m/mongosh.rb#L24)
- [`code-server`](https://github.com/Homebrew/homebrew-core/blob/f4861d055bf1593f5d0b1934b4756cb9ed9e8246/Formula/c/code-server.rb#L48)
- [`bcoin`](https://github.com/Homebrew/homebrew-core/blob/f4861d055bf1593f5d0b1934b4756cb9ed9e8246/Formula/b/bcoin.rb#L36)
- [`emscripten`](https://github.com/Homebrew/homebrew-core/blob/f4861d055bf1593f5d0b1934b4756cb9ed9e8246/Formula/e/emscripten.rb#L187)
  - This is another case where the formula is much more complex than the sample here os it probably shouldn't be included

On the other hand, there are 155 formulae that use `bin.install_symlink` and 145 of those use `bin.install_symlink Dir[libexec/"foo"]`. So, the original way this is written seems to be by far the most prevalent.

Reply:
Thanks @Rylan12!

> On the other hand, there are 155 formulae that use `bin.install_symlink` and 145 of those use `bin.install_symlink Dir[libexec/"foo"]`. So, the original way this is written seems to be by far the most prevalent.

Given the prevalence here: I think this may be worth opening a Homebrew/core issue to discuss if we want to migrate the approach here. I don't think it makes sense to have the documentation recommend the opposite of what most formulae are doing.

Reply:
I opened it here – https://github.com/Homebrew/homebrew-core/issues/176257

Reply:
That's perfect, thanks!